Pancakeswap-Agents/LargeLPTokenDepositWithdraw/src/agent.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  Finding,
+  HandleTransaction,
+  TransactionEvent,
+  getEthersProvider,
+} from "forta-agent";
+
+import { BigNumber, ethers } from "ethers";
+import {
+  MASTERCHEF_ADDRESS,
+  MASTERCHEF_ABI,
+  IBEP20_ABI
+} from './constants';
+
+import { BotConfig, STATIC_CONFIG, DYNAMIC_CONFIG } from "./config";
+import { createFinding } from "./findings";
+
+
+export function provideHandleTransaction(
+  config: BotConfig,
+  provider: ethers.providers.Provider
+) : HandleTransaction {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+      
+    let findings: Finding[] = [];
+
+    const masterchefContract = new ethers.Contract(MASTERCHEF_ADDRESS, MASTERCHEF_ABI, provider);
+
+    await Promise.all(
+      // filter the transaction logs for Deposit and withdraw events
+      txEvent.filterLog(
+        [MASTERCHEF_ABI[0], MASTERCHEF_ABI[1]],
+        MASTERCHEF_ADDRESS
```

Comment:
Here is better to use an address passed as a parameter instead of a global one. That will allow you to use a different one in the tests.